File: PhotographySessionLogger/src/PhotographySessionLogger.Core/Model/ProjectAggregate/Project.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace PhotographySessionLogger.Core;

public class Project
{
    public Guid ProjectId { get; set; }
    public Guid UserId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime? DueDate { get; set; }
    public bool IsCompleted { get; set; }
    public string? Notes { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: HomeImprovementProjectManager/src/HomeImprovementProjectManager.WebApp/projects/home-improvement-project-manager/src/app/models/project.ts
import { ProjectStatus } from './project-status';

export interface Project {
  projectId: string;
  userId: string;
  name: string;
  description?: string;
  status: ProjectStatus;
  startDate?: string;
  endDate?: string;
  estimatedCost?: number;
  actualCost?: number;
  createdAt: string;
}
